sparse/core.py

```diff
@@ -8,12 +8,21 @@
 import numpy as np
 import scipy.sparse
 
+# zip_longest with Python 2/3 compat
+try:
+    from itertools import zip_longest
+except ImportError:
+    from itertools import izip_longest as zip_longest
 
 try:  # Windows compatibility
     int = long
 except NameError:
     pass
 
+# Generated once with np.random.rand(). Used for checking if an elemwise function returns zero for specific inputs.
+_TEST_NONZERO = 0.8173885343707892
+_TEST_NONZERO_2 = 0.32621130891611305
```

Comment:
I needed your input here. Right now, I'm just checking against random values if the function is nonzero for specific combinations. I could check with `(func(self.data, 0, *args, **kwargs) == 0).all()` and `(func(0, other.data, *args, **kwargs) == 0).all`. Do you think that's better? (It's algorithmically correct, but can take longer).

Alternately we can store these results somewhere and use them.

Reply:
This seems fragile to me.  For example what happens if the function has different behavior for positive and negative numbers?